### PR TITLE
Smart Contract IA

### PR DESCRIPTION
**Pull request by crazeekennee**
## What/Why/How?

This PR introduces a new Information Architecture that subsequent categories will follow, and applies it to the Smart Contract category.

## Screenshots

Screenshot 2023-03-01 at 2 59 53 PM

https://user-images.githubusercontent.com/13068878/222253967-4f9637d0-14b7-46f3-9d8f-c9c2a01df429.mov

## Check yourself

- [x] Code is linted
- [x] Tested
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: smart-contracts/sidebars.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+pages:
+  - separator: Smart Contract
```
Comment (sammyluo) [inline smart-contracts/sidebars.yaml:2]:
```suggestion
  - separator: Smart Contracts
```

File: smart-contracts/sidebars.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  - group: Overview
+    expanded: false
+    pages:
+      - label: UNS Architecture
+        page: ../getting-started/domain-registry-essentials/uns-architecture-overview.md
+      - label: CNS Architecture
+        page: ../getting-started/domain-registry-essentials/cns-architecture-overview.md
+      - label: UNS vs CNS comparison
+        page: ../getting-started/domain-registry-essentials/cns-uns-comparison.md
+  - group: Quick Start
```
Comment (sammyluo) [inline smart-contracts/sidebars.yaml:13]:
```suggestion
  - group: Quickstart
```

**Review by sammyluo — CHANGES_REQUESTED** [review]
Let's also update the folder structure we discussed.

File: smart-contracts/quick-start/manage-domain-records.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
+---
+title: Manage Domains Using Smart Contracts Guide | UD Developer Portal
+description: This guide covers how to manage UD domain records using the Etherscan and Polygonscan user interfaces to write and execute proxy contracts.
+redirectFrom:
+  - /manage-domains/managing-domains-using-contracts/
+---
+
+# Manage Domains Using Smart Contracts Guide
```
Comment (sammyluo) [inline smart-contracts/quick-start/manage-domain-records.md:8]:
```suggestion
# Manage Domains Using Smart Contracts
```

File: smart-contracts/quick-start/mint-subdomains.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+---
+title: Mint Subdomains Using Smart Contracts Guide | UD Developer Portal
+description: This guide covers how to mint UD subdomains using the Polygonscan user interface to write and execute proxy contracts.
+redirectFrom:
+  - /manage-domains/mint-subdomains-using-contracts/
+---
+
+# Mint Subdomains Using Smart Contracts Guide
```
Comment (sammyluo) [inline smart-contracts/quick-start/mint-subdomains.md:8]:
```suggestion
# Mint Subdomains Using Smart Contracts
```

File: smart-contracts/quick-start/resolve-domains.md [hunk 5]
```diff
@@ -0,0 +1,479 @@
+---
```
Comment (sammyluo) [inline smart-contracts/quick-start/resolve-domains.md:1]: Resolve domains should be this one: https://docs.unstoppabledomains.com/developer-toolkit/resolution-integration-methods/direct-blockchain-calls/resolve-eth-smart-contracts/

File: smart-contracts/quick-start/set-reverse-records.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
+---
+title: Smart Contract Configuration Guide | UD Developer Portal
+description: This guide covers how to set Reverse Resolution records for UD domains using smart contracts. This process requires using the ABIs implemented into the Unstoppable Domains UNS smart contract.
+---
+
+# Smart Contract Configuration Guide
```
Comment (sammyluo) [inline smart-contracts/quick-start/set-reverse-records.md:6]:
```suggestion
# Set Reverse Records Using Smart Contracts
```

File: smart-contracts/quick-start/reverse-resolve-domains.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
+---
+title: Smart Contract Integration Guide | UD Developer Portal
+description: This guide covers how to retrieve the reverse record of UD domains using smart contracts. This process requires using the ABIs built into the Unstoppable Domains UNS smart contract.
+---
+
+# Smart Contract Integration Guide
```
Comment (sammyluo) [inline smart-contracts/quick-start/reverse-resolve-domains.md:6]:
```suggestion
# Reverse Resolve Domains Using Smart Contracts
```

**Review by sammyluo — CHANGES_REQUESTED** [review]

File: smart-contracts/quick-start/resolve-domains.md (hunk 5 above)
Comment (crazeekennee) [reply smart-contracts/quick-start/resolve-domains.md:1]: done

File: smart-contracts/quick-start/resolve-domains.md [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+title: Resolve Using Smart Contracts Guide | UD Developer Portal
+description: This page explains the process for resolving domain records by making calls to Ethereum and Polygon smart contracts using Ethereum JSON RPC.
+---
+
+# Resolve Domains Using Smart Contracts Guide
```
Comment (sammyluo) [inline smart-contracts/quick-start/resolve-domains.md:6]:
```suggestion
# Resolve Domains Using Smart Contracts
```

**Review by sammyluo — APPROVED** [review]
Nice!

**Pull request merged**